packages/muelu/CMakeLists.txt

```diff
@@ -265,8 +265,10 @@ GLOBAL_SET(HAVE_${PACKAGE_NAME_UC}_TEKO ${${PACKAGE_NAME}_ENABLE_Teko})
 # - MueLu_ENABLE_EXPLICIT_INSTANTIATION
 # - Enabling/disabling of scalar/ordinal types is controlled by Tpetra
 
-IF(${PACKAGE_NAME}_ENABLE_EXPLICIT_INSTANTIATION)
+GLOBAL_SET(HAVE_${PACKAGE_NAME_UC}_DEFAULT_GO_LONG OFF)
+GLOBAL_SET(HAVE_${PACKAGE_NAME_UC}_DEFAULT_GO_LONGLONG OFF)
 
+IF(${PACKAGE_NAME}_ENABLE_EXPLICIT_INSTANTIATION)
```

Comment:
The set of enabled types should not depend on whether explicit instantiation is enabled.  For example, Belos uses the set of enabled types for SolverFactory setup, and many packages use the set of enabled types to decide what tests to enable.

Reply:
@mhoemmen This doesn't affect what's enabled or instantiated, just what is used to pick one type as the default template parameter. The user of MueLu_blah.hpp can still put anything they want as the GO template parameter.

Reply:
@mhoemmen This was all supposed to go into #5697 which removed the default SC,LO,GO from Xpetra entirely, and tried to change MueLu so that the default GO template parameter was chosen from the set of enabled GOs, since before both Xpetra and MueLu used int no matter what.